examples/gocounter/go.mod

```diff
@@ -10,15 +10,40 @@ require (
 	github.com/cilium/ebpf v0.9.3
 	github.com/redhat-et/bpfd/clients/gobpfd v0.0.0-20221213142718-d9b708634d05
 	github.com/redhat-et/bpfd/examples/pkg/config-mgmt v0.0.0-00010101000000-000000000000
+	github.com/redhat-et/bpfd/examples/pkg/dyn-client v0.0.0-00010101000000-000000000000
 	google.golang.org/grpc v1.51.0
 )
 
 require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/go-logr/logr v1.2.3 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/google/gofuzz v1.1.0 // indirect
+	github.com/json-iterator/go v1.1.12 // indirect
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
+	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/pelletier/go-toml v1.9.5 // indirect
-	golang.org/x/net v0.0.0-20220722155237-a158d28d115b // indirect
-	golang.org/x/sys v0.0.0-20220928140112-f11e5e49a4ec // indirect
-	golang.org/x/text v0.4.0 // indirect
+	github.com/redhat-et/bpfd/bpfd-operator v0.0.0-20221213225838-ca31b67e44d2 // indirect
+	golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10 // indirect
+	golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b // indirect
+	golang.org/x/sys v0.3.0 // indirect
+	golang.org/x/term v0.3.0 // indirect
+	golang.org/x/text v0.5.0 // indirect
+	golang.org/x/time v0.0.0-20220609170525-579cf78fd858 // indirect
+	google.golang.org/appengine v1.6.7 // indirect
 	google.golang.org/genproto v0.0.0-20220502173005-c8bf987b8c21 // indirect
-	google.golang.org/protobuf v1.28.0 // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/apimachinery v0.26.0 // indirect
+	k8s.io/client-go v0.26.0 // indirect
+	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/utils v0.0.0-20221107191617-1a15be271d1d // indirect
+	sigs.k8s.io/controller-runtime v0.13.0 // indirect
+	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/redhat-et/bpfd/examples/pkg/dyn-client => ../pkg/dyn-client
```

Comment:
So we aren't planning for external consumers to use this package right? if not I would just have a single go.mod/ go.sum for the entire examples directory so then we don't need any funny hacks like this to pickup local changes Ultimately I think it should look like 

```
examples/
    go.mod
    go.sum
    internal/ 
        - Contains shared libaries for examples/ 
    go-tc-counter/ 
         - main.go 
         - etc...
    go-xdp-counter/
         - main.go
         - etc
    go-tracepoint/
    etc...
 ```

Reply:
Then a user can simply cd examples and do `go run --exec <EXAMPLE` similar to [how cilium organizes things](https://github.com/cilium/ebpf/tree/master/examples)

Reply:
Lol you did this in a later commit :)

Reply:
I think just squashing 5a671e7955538d92e36d63a826ce4633f1092954 bd0fb164bf47558c84c8fe9c099c5c4c93348ab1 and 394d05cf1c558d8854208f47548f82aa9434ab84 together will help :)

Reply:
Done